policies/definitions.md

```diff
@@ -0,0 +1,95 @@
+# OpenSSL terms and definitions
+
+## ABI
+
+Application binary interface.
+
+An ABI compatible release allows applications to
+be relinked against the new library with the expectation that everything will
+continue to function.
+
+## Alpha release
+
+An [alpha release] is an early pre-release version.
+Refer to the [versioning policy] for specific details.
+
+## API
+
+Application programming interface.
+
+An API compatible release allows application to be recompiled against the new
+library with the expectation that everything will continue to function.
+
+Refer to the [API compatibility policy] for specific details.
+
+## Beta release
+
+A [beta release] is a late pre-release version.
+Refer to the [versioning policy] for specific details.
+
+## Bug fix
+
+A bug fix is a fix of functionality of the libraries, modules, applications
+or the build system.
+Refer to the [stable release update policy] for specific details.
+
+## End-user documentation
+
+End-user documentation is documentation intended for users of the OpenSSL
+libraries and commandline utilities.
+Refer to the [stable release update policy] for specific details.
+
+## LTS
+
+[Long term support].  These releases are supported for longer than normal
+releases.  Refer to the [versioning policy] for specific details.
+
+## Major release
+
+A [major release] is one where API and ABI breaking changes are permitted.
+Refer to the [versioning policy] for specific details.
+
+## Minor release
+
+A [minor release] is one where API and ABI breaking changes are **not** permitted
+and additional functionality can be added.
+Refer to the [versioning policy] for specific details.
+
+## OMC
+
+_OpenSSL Management Committee_.
+This group oversees all managerial and administrative aspects of the project.
+The OMC is the final authority for the OpenSSL project.
+
+## OTC
+
+_OpenSSL Technical Committee_.
+This group oversees all technical aspects of the project.
+
+## Patch release
+
+A [patch release] is one where API and ABI breaking changes are **not** permitted
+and no additional functionality can be added.
+Refer to the [versioning policy] for specific details.
+
+## Public interface
+
+A public interface is any function, global variable, structure or macro
+declared in a public header file.
+Refer to the [API compatibility policy] for specific details.
+
+## Stable release
+
+A stable release is a patch release from an existing supported minor
+release branch.
```

Comment:
Not sure I understand how this differs from "Patch release", defined above.

Also, I think that we have commonly called this a "Release update"...

Reply:
We use the term _stable release_ in our policies, it needs to be defined.  Here is where it is defined.

The document, perhaps, should also define _release update_.  We don't appear to have used that term (yet) in the policy repos.

Reply:
Yeah, and actually, this definition isn't quite it.  A stable release *isa* minor or major release (including all updates, present and future), i.e. 3.0.0 is a stable release (or one could say 3.0.x).

Reply:
This is the definition from our [stable release policy](https://github.openssl.org/otc/technical-policies/blob/master/policies/stable-release-updates.md#definitions), which I'll note you voted in favour of :grin:

3.0.0 is technically a patch release (as well as a minor and a major one).  From this perspective, the statement is correct.

Reply:
Damn it, I didn't notice it then. I find this description quite confusing, and your attempt at retrofitting an explanation doesn't help.

Reply:
Suggest what the glossary should say about it.

We can and should more rigorously define the term in the stable release policy.

Reply:
Fixed it here.

Reply:
I've submitted an amendment PR for the stable release policy.  openssl/technical-policies#24